trainee_template/ablation.amlg

```diff
@@ -0,0 +1,363 @@
+                )
+        )
+
+        (call StoreCaseValues (assoc
+            case_values_map influence_weight_entropy_map
+            label_name
+                (if
+                    (= (null) label_name)
+                    internalLabelInfluenceWeightEntropy
+                    label_name
+                )
+        ))
+
+        (accum_to_entities (assoc revision 1))
+        (assign_to_entities (assoc hasInfluenceWeightEntropies (true)))
+    )
+
+    ;initialize parameters and internal features related to auto ablation
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #InitializeAutoAblate
+    (declare
+        (assoc
+            weight_feature autoAblateWeightFeature
+        )
+
+        (seq
+            (assign_to_entities (assoc hasPopulatedCaseWeight (true)))
+            (accum_to_entities (assoc revision 1))
+            (call CreateCaseWeights (assoc
+                feature_name weight_feature
+            ))
+        )
+        
+    )
+
+    ;determine whether a new case (one that is not in the Trainee) should be ablated (trained as weights) or kept.
+    ; context_values: list of context values to react to
+    ; context_features: list of context features to react to
+    #ShouldNewCaseBeAblated
+    (declare
+        (assoc
+            context_features (list)
+            context_values (list)
+
+            ;internal
+            max_influence_weight_entropy_to_keep .infinity
+        )
+
+        ;If we do not have influence weight entropies stored, ablation cannot happen.
+        ; So, always return 1 when that is the case.  The same is true for when there
+        ; are not enough cases in the model.
+        (if
+            (or 
+                (not (and hasInfluenceWeightEntropies))
+                (< (call GetNumTrainingCases) autoAblateMinModelSize )
+            )
+            (conclude 1)
+        )
+
+        (assign (assoc
+            max_influence_weight_entropy_to_keep
+                (compute_on_contained_entities (list
+                    (query_exists internalLabelInfluenceWeightEntropy)
+                    (query_quantile internalLabelInfluenceWeightEntropy autoAblateInfluenceWeightEntropyThreshold ".case_weight")
+                ))
+        ))
+
+        (+
+            (or
+                (> max_influence_weight_entropy_to_keep
+                    (call ComputeInfluenceWeightEntropy (assoc
+                        context_values context_values
+                        context_features context_features
+                    ))
+                    
+                )
+                ; This is probably unnecessary but in case we cannot compute the requested influence weight
+                ; quantile return 1 as well.
+                (= max_influence_weight_entropy_to_keep .nan)
+            )
+        )
+    )
+
+    #CaseOutsideThresholds
+	(let
+		(assoc
+			action_features
+                (filter (lambda
```

Comment:
maybe the github formatting is off, but I think the indentation here needs to be tweaked. Multi-op values of an assoc should be on a new line and indented